lib/fizzy/execute.cpp

```diff
@@ -368,9 +447,10 @@ Instance instantiate(const Module& module, std::vector<ImportedFunction> importe
 
         // NOTE: these instructions can overlap
         assert((offset + data.init.size()) <= (memory_max * PageSize));
-        std::memcpy(memory.data() + offset, data.init.data(), data.init.size());
+        std::memcpy(memory->data() + offset, data.init.data(), data.init.size());
     }
 
+    // NOLINTNEXTLINE(clang-analyzer-cplusplus.NewDeleteLeaks)
```

Comment:
I could format it in a way to have `NOLINT` only on a single move, but weirdly the warning itself points to moving of a `table` (while the allocation in question is the one of `memory`)
So this would help
```
    Instance instance = {module, std::move(memory), memory_max,
        std::move(table),  // NOLINT(clang-analyzer-cplusplus.NewDeleteLeaks)
        std::move(globals), std::move(imported_functions), std::move(imported_function_types),
        std::move(imported_globals)};

```
but not this:
```
    Instance instance = {module,
        std::move(memory),  // NOLINT(clang-analyzer-cplusplus.NewDeleteLeaks)
        memory_max, std::move(table), std::move(globals), std::move(imported_functions),
        std::move(imported_function_types), std::move(imported_globals)};
```

Reply:
Is this the same bug? https://bugs.llvm.org/show_bug.cgi?id=32234

Reply:
I don't know, probably it's not. I can't really reproduce it outside of this our code.

Reply:
How it is done right now is fine, but add a FIXME comment to investigate later (I'm willing to take care of it).